src/Microsoft.Framework.Configuration/ConfigurationKeyComparer.cs

```diff
@@ -8,14 +8,12 @@ namespace Microsoft.Framework.Configuration
 {
     public class ConfigurationKeyComparer : IComparer<string>
     {
-        private const char Separator = ':';
-
         public static ConfigurationKeyComparer Instance { get; } = new ConfigurationKeyComparer();
 
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(Separator) ?? new string[0];
-            var yParts = y?.Split(Separator) ?? new string[0];
+            var xParts = x?.Split(new string[]{ Constants.KeyDelimiter }, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
```

Comment:
The method that matches on `Split('x')` really just has a `params` parameter to get the nice syntax, but there's an array allocation there anyway. I'd just use the explicit array overload and avoid the LINQ statement which kind of makes the code unreadable.